### ISSUE TITLE
Add ToggleSwitch control

### ISSUE THREAD

**Issue by harinikmsft** [Microsoft/xaml-standard#17]
```xml
<ToggleSwitch x:Name="BackupToggle"
     IsOn="{Binding Path=MyViewModel.IsBackupOn, Mode=TwoWay}"
     Toggled="BackupToggle_Toggled" />
```

**Comment by insinfo** [comment]
Very used

**Comment by h82258652** [comment]
also ```OnContent```, ```OffContent``` property.

**Comment by nickrandolph** [comment]
@h82258652 by OnContent, OffContent, are you referring to how each state is represented visually? If so, this should be expressed in the template for the control.

**Comment by birbilis** [comment]
means a content control with two visual subtrees obviously, also see bit heated discussion on CheckBox at https://github.com/Microsoft/xaml-standard/issues/170

ToggleButton also mentioned there near the end - I suggest a N-state (aka generic) ToggleButton in fact if we're going to have one that is separate from a three-state (yep, with optional undecided state) CheckBox

after all, why should Button be a content control and other controls that have button-like interaction shouldn't?

**Comment by nickrandolph** [comment]
But if the control template for the ToggleSwitch could be overridden, this would remove the need to have two visual subtrees.

**Comment by birbilis** [comment]
Do you mean the UWP button doesn't have a Content property?

Visual states is a more verbose approach that one should be able to choose or refactor it later on if they wish to. Especially when porting UI from other (esp. earlier like WinForms) platforms

**Comment by birbilis** [comment]
The alternative could be to have more simple controls as a separate library (PlainButton etc.) that underneath use the more verbose approach. However, I'd prefer to have (multi)content controls that define the visual tree structure (branches) with their properties. The properties are the meta-information of the UI structure that is, not having it hidden from the programmatic layer in some opaque (or one that needs magic strings like part names or introspection etc. to access) template

**Comment by dotMorten** [comment]
IMHO the N content idea here is just as wrong as it was over in the checkbox discussion. Toggle button already has an expected behavior and there's no intent to break years of accepted behavior or break uwp. 
"Toggle" really conveys flipping between two values optionally with an unknown state

**Comment by birbilis** [comment]
as you mentioned at that other thread, Checkbox is sometimes implemented by descending from a ToggleButton. So despite you not wanting three-state checkboxes, most frameworks have such. So in that case you need N-state semantics.

also, clicking to toggle between a sequence of states isn't something that is unheard of I believe. Was there long ago before touch gestures appeared.

**Comment by nickrandolph** [comment]
I agree with @dotMorten the N content is not the way to go. Content w ContentTemplate is the preferred mechanism for all controls as this means that developers can override and customise to hearts content. I don't think we're disagreeing that there needs to be the ability to shift through N-states but this can easily be handled with the visual state mechanism that's supported by UWP, WPF and Silverlight (and yes a *major* weakness of Forms imho)

**Comment by birbilis** [comment]
One could use a templated control to make anything - in that sense you'd mostly need templated controls and no behaviour specified on the control API (PME model - Property/Method/Event - in fact this repo seems to just specify the P and E, not M). 

But this isn't a component-based UI. A component should expose its behaviour in a contract that is easily accessible/understandable via the Property/Event editor and via the object browser, not assume the programmer will dig to find the template and examine it. Nor should the programmer have to rely on hardcoded magic strings (part names). Those part names should be defined in the codebehind and liked to properties, then templates should use those part names so that the properties can find those parts.

**Comment by birbilis** [comment]
...and ideally those properties should be of some interface (see https://github.com/Microsoft/xaml-standard/issues/95) type (which again seems to not be the focus of this repo)

**Comment by birbilis** [comment]
btw, the 2-state toggle is related to FlipPanel control too, that do a transition between two UIs (a front and a back)

See http://clipflair.codeplex.com/SourceControl/latest#Client/ZUI/FlipPanel/ for example. That code can be seen in action at http://studio.clipflair.net (visit with IE or can open IE from Edge, needs Silverlight), where each component (including the zoomable container which is a component too that can be placed inside another container etc.) has a wrench button on its titlebar to flip it over and see its property pages. After editing those can flip over again to continue using the component's UI. Using a simple sliding transition there, another one in that code was a more 3D-like flipping effect

Speaking of card flipping, and since I wrote about N-state toggling above, in cards that would be like having a card deck (like those card organizers found on desks in the past ) and rolling to the next card etc. till the last one. With a palindrome option you'd roll back to the first one eventually. So it would be more of a generalization of the 2-faced card flipping
